rclcpp/test/rclcpp/topic_statistics/test_subscription_topic_statistics.cpp

```diff
@@ -318,10 +318,27 @@ TEST_F(TestSubscriptionTopicStatisticsFixture, test_receive_stats_for_message_no
   // Check the collected statistics for message period.
   // Message age statistics will not be calculated because Empty messages
   // don't have a `header` with timestamp.
+  bool any_samples = false;
   for (const auto & msg : received_messages) {
     if (msg.metrics_source != "message_period") {
       continue;
     }
+    // skip messages without samples
```

Comment:
Nit: could make this as a separate function (I should have done the same for some of the reused code) for the multiple tests.

Reply:
Note: If you receive 4 messages I would expect that 2 of them are the `message_age` measurement and the remaining 2 are the `message_period` measurement. Message period is calculated on the subscriber side, whereas [message age requires that message timestamp is present](https://index.ros.org/doc/ros2/Concepts/About-Topic-Statistics/#types-of-statistics-calculated). From the [behavior](https://index.ros.org/doc/ros2/Concepts/About-Topic-Statistics/#behavior) notice how the [sample count is zero](https://index.ros.org/doc/ros2/Tutorials/Topics/Topic-Statistics-Tutorial/#observe-published-statistic-data), so that would denote the `message_age` measurement. 

My guess is this test was broken from the start because it didn't attempt to check the `message age` case, but only the `message period`. What we could do here instead is simply add a count for each type of measurement received and assert false otherwise, e.g., your 

```
    if (!has_samples) {
      continue;
    }
```

would just turn into checking a counter at the end instead of the boolean flag.

Reply:
Both suggestions would be good for improving the test but for the scope of this PR which just aims to fix the failure with Connext I would rather not do those.

Reply:
I will pick up adding the improvement. I was also incorrect regarding the counts above: if you receive 4 messages, I would expect 4 `message age` and 4 `message period` messages for a total of 8 statistics messages published: 1 for each of of the messages recieved.